File: Runtime/Plugins/unity.androidlib/src/main/java/com/affise/attribution/unity/common/MethodResult.java
package com.affise.attribution.unity.common;

public class MethodResult implements Result {

    private Object result;
    private MethodError error;
    private boolean notImplemented = false;

    public void success(Object result) {
        this.result = result;
    }

    public void error(String errorCode, String errorMessage, Object errorDetails) {
        this.error = new MethodError(errorCode, errorMessage, errorDetails);
    }

    public void notImplemented() {
        notImplemented = true;
    }

    public Object getResult() {
        return result;
    }

    public MethodError getError() {
        return error;
    }

    public boolean isError() {
        return error != null;
    }

    public boolean isNotImplemented() {
        return notImplemented;
    }
}

File: Runtime/Plugins/iOS/Source/common/MethodResult.swift
public class MethodResult : NativeResult {
    
    private var result: Any?
    private var error: MethodError?
    private var implemented: Bool = false
    
    func success(_ result: Any?) {
        self.result = result
    }
    
    func error(_ message: String) {
        self.error = MethodError(message: message)
    }
    
    func error(_ errorCode: String, errorMessage: String? = nil, errorDetails: Any? = nil) {
        self.error = MethodError(errorCode: errorCode, errorMessage: errorMessage, errorDetails: errorDetails)
    }
    
    func notImplemented() {
        implemented = true;
    }
    
    func getResult() -> Any? {
        return result
    }
    
    func getError() -> MethodError? {
        return error
    }
    
    func isError() -> Bool {
        return error != nil
    }
    
    func isNotImplemented() -> Bool {
        return implemented
    }
}
